enterprise/frontend/src/embedding-sdk/components/public/NewQuestion/NewQuestion.tsx

```diff
@@ -0,0 +1,136 @@
+import { Button, Group, Stack, Tabs, Title } from "metabase/ui";
+
+import {
+  Notebook,
+  QuestionVisualization,
+} from "../../private/InteractiveQuestion/components";
+import {
+  InteractiveQuestionProvider,
+  useInteractiveQuestionContext,
+} from "../../private/InteractiveQuestion/context";
+
+const SaveQuestion = ({ onClose }: { onClose: () => void }) => {
+  const { question, originalQuestion } = useInteractiveQuestionContext();
+
+  const handleCreate = useCreateQuestion();
+  const handleSave = useSaveQuestion();
+
+  if (!question) {
+    return null;
+  }
+
+  return (
+    <SaveQuestionProvider
+      question={question}
+      originalQuestion={originalQuestion ?? null}
+      onCreate={handleCreate}
+      onSave={handleSave}
+      multiStep={false}
+      initialCollectionId={null}
+    >
+      <Stack p="md">
+        <Title>
+          <SaveQuestionTitle />
+        </Title>
+        <SaveQuestionForm onCancel={onClose} />
+      </Stack>
+    </SaveQuestionProvider>
+  );
+};
+
+const ResetQuestionButton = ({ onClick }: { onClick?: () => void } = {}) => {
+  const { onReset } = useInteractiveQuestionContext();
+
+  const handleReset = () => {
+    onReset?.();
+    onClick?.();
+  };
+
+  return <Button onClick={handleReset}>Reset</Button>;
+};
+
+const NewQuestionInner = () => {
+  const [activeTab, setActiveTab] = useState<
+    "notebook" | "visualization" | "save" | null | (string & unknown)
+  >("notebook");
+
+  const { queryResults } = useInteractiveQuestionContext();
+
+  const previousTab = usePreviousDistinct(activeTab);
+
+  const returnToPreviousTab = () => {
+    if (previousTab) {
+      return setActiveTab(previousTab);
+    }
+  };
+
+  const onSaveButtonClick = () => {
+    if (activeTab === "save") {
+      returnToPreviousTab();
+    } else {
+      setActiveTab("save");
+    }
+  };
+
+  return (
+    <Tabs
+      value={activeTab}
+      onTabChange={setActiveTab}
+      style={{
+        // we have to use a style tag because Mantine uses inline styles for defining flex direction
+        flexDirection: "column",
+      }}
+      display="flex"
+      h="100%"
+    >
+      <Group w="100%" position="apart">
+        <Tabs.List>
+          <Group>
+            <Tabs.Tab value="notebook">Notebook</Tabs.Tab>
+            {queryResults && (
+              <Tabs.Tab value="visualization">Visualization</Tabs.Tab>
+            )}
+          </Group>
+        </Tabs.List>
+        {activeTab !== "save" && (
+          <Group>
+            <ResetQuestionButton onClick={() => setActiveTab("notebook")} />
+            {/* using a button instead of a tab for styling reasons */}
+            <Button onClick={onSaveButtonClick}>Save</Button>
+          </Group>
+        )}
+      </Group>
+
+      <Tabs.Panel value="notebook">
+        <Notebook onApply={() => setActiveTab("visualization")} />
+      </Tabs.Panel>
+      <Tabs.Panel value="visualization" style={{ flex: 1 }}>
```

Comment:
Adding this allows the visualization to take up the whole element size. For whatever reason, the Tabs component in Mantine doesn't actually provide the child components with any sort of height value (it's not even `100%`, but just not set at all). Otherwise, the visualization will be cut off at 500px all the time, which will definitely not be something that clients will want 100% of the time